sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceEvaluator.java

```diff
@@ -0,0 +1,873 @@
+            return false;
+        }
+
+        // Check whether "Start" is a valid first occurrence
+        final ZonedDateTime alignedStart = settings.getStart().withZoneSameInstant(getRecurrenceTimeZoneId());
+        if (alignedStart.getMonthValue() != pattern.getMonth()
+            || pattern.getDaysOfWeek().stream().noneMatch(day ->
+                alignedStart.isEqual(dayOfNthWeekInTheMonth(alignedStart, pattern.getIndex(), day)))) {
+            paramName = TIME_WINDOW_FILTER_SETTING_START;
+            reason = RecurrenceConstants.NOT_MATCHED;
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Find the nth day of week in the month.
+     *
+     * @param startDateTime start date time, specifies the month number
+     * @param index         Specifies on which instance of the allowed days specified in daysOfsWeek the event occurs, counted from the first instance in the month
+     * @param dayOfWeek     Specifies on which day of the week the event can occur
+     * @return The data time of the day in nth week the month.
+     */
+    private ZonedDateTime dayOfNthWeekInTheMonth(ZonedDateTime startDateTime, String index, String dayOfWeek) {
+        ZonedDateTime dateTime = ZonedDateTime.of(startDateTime.getYear(), startDateTime.getMonthValue(), 1,
+            startDateTime.getHour(), startDateTime.getMinute(), startDateTime.getSecond(), startDateTime.getNano(), startDateTime.getZone());
+
+        // Find the first day of week in the month
+        while (getDayOfWeek(dateTime) != convertToWeekDayNumber(dayOfWeek)) {
+            dateTime = dateTime.plusDays(1);
+        }
+
+        final ZonedDateTime tempDateTime = dateTime.plusDays((long) RecurrenceConstants.WEEK_DAY_NUMBER * (convertIndexToNumber(index) - 1));
+        if (index.equalsIgnoreCase(RecurrenceConstants.LAST) && tempDateTime.getMonthValue() != startDateTime.getMonthValue()) {
+            // There is no the 5th week in the month
+            // Add 3 weeks to reach the fourth week in the month
+            dateTime = dateTime.plusDays(RecurrenceConstants.WEEK_DAY_NUMBER * 3);
+        } else {
+            dateTime = dateTime.plusDays((long) RecurrenceConstants.WEEK_DAY_NUMBER * (convertIndexToNumber(index) - 1));
+        }
+        return dateTime;
+    }
+
+    private boolean tryValidateDaysOfWeek(TimeWindowFilterSettings settings) {
+        paramName = String.format("%s.%s.%s", TIME_WINDOW_FILTER_SETTING_RECURRENCE, RecurrenceConstants.RECURRENCE_PATTERN,
+            RecurrenceConstants.RECURRENCE_PATTERN_DAYS_OF_WEEK);
+        final List<String> daysOfWeek = settings.getRecurrence().getPattern().getDaysOfWeek();
+        if (daysOfWeek == null || daysOfWeek.size() == 0) {
+            reason = RecurrenceConstants.REQUIRED_PARAMETER;
+            return false;
+        }
+
+        final String[] legalValues = {RecurrenceConstants.MONDAY, RecurrenceConstants.TUESDAY, RecurrenceConstants.WEDNESDAY,
```

Comment:
`DayOfWeek[] legalValues = DayOfWeek.values();` will give you a list of all of these as enums. And using `.name()` gets you the string value.